### PR TITLE
widelands: update to 1.1.

### PR DESCRIPTION
**Pull request by kruceter**
cc: @travankor 

Related to #39083.

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-musl
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - x86_64
  - armv6l-musl (cross)

### CODE REVIEW

File: srcpkgs/widelands/patches/wordsize.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
+--- a/src/base/macros.h
++++ b/src/base/macros.h
+@@ -28,6 +28,11 @@
+ #endif
+ #endif
+
++/* Used for wordsize only */
++#if !defined(_WIN32) && !defined(_WIN64)
++#include <limits.h>
++#endif
```
Comment (sgn) [inline srcpkgs/widelands/patches/wordsize.patch:10]: Why not include it unconditionally?

File: srcpkgs/widelands/patches/wordsize.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  * From http://dbp-consulting.com/tutorials/SuppressingGCCWarnings.html and
+  * slightly modified to remove support entirely for GCC < 4.6 because we'll
+@@ -135,7 +140,7 @@
+ #define PRIuS PRIu32
+ #endif
+ #else
+-#if __WORDSIZE == 64
++#if (ULONG_MAX == 0xffffffffffffffff)
+ #define PRIuS "lu"
+ #else
```
Comment (sgn) [inline srcpkgs/widelands/patches/wordsize.patch:22]: You need to have a patch that highlight 32bit platform, too.

Comment (kruceter) [reply srcpkgs/widelands/patches/wordsize.patch:20]: Not marking as resolved yet. I hope `#if (ULONG_MAX == 0xffffffff)` is the right directive for 32 bit platforms.

Comment (sgn) [reply srcpkgs/widelands/patches/wordsize.patch:20]: Hm, it can be simple than that, it's `zu`

**Comment by sgn** [comment]
Merged.

**Pull request closed without merging**